tests/foreman/performance/test_candlepin_concurrent_subscription_attach.py

```diff
@@ -0,0 +1,116 @@
+"""Test class for concurrent subscription by register and attach"""
+from datetime import date
+from robottelo.common import conf
+from robottelo.cli.subscription import Subscription
+
+from robottelo.performance.perf_concurrent_test import ConcurrentTestCase
+
+
+class ConcurrentSubAttachTestCase(ConcurrentTestCase):
+    """Concurrent Subscribe to Red Hat Satellite 6 Server by attach tests"""
+
+    @classmethod
+    def setUpClass(cls):
+        super(ConcurrentSubAttachTestCase, cls).setUpClass()
+
+        # parameters for concurrent register and attach test
+        cls.savepoint = conf.properties[
+            'performance.test.savepoint2_enabled_repos']
+        cls.raw_file_name = conf.properties[
+            'performance.csv.raw_att_file_name']
+        cls.stat_file_name = conf.properties[
+            'performance.csv.stat_att_file_name']
+
+        # add date to csv files names
+        today = date.today()
+        cls.raw_file_name = '{0}-{1}'.format(
+            today.strftime('%Y%m%d'), cls.raw_file_name)
+        cls.stat_file_name = '{0}-{1}'.format(
+            today.strftime('%Y%m%d'), cls.stat_file_name)
+
+    def setUp(self):
+        super(ConcurrentSubAttachTestCase, self).setUp()
+
+        # Restore database before concurrent subscription
+        self._restore_from_savepoint(self.savepoint)
+
+        # Get subscription id
+        (self.sub_id, sub_name) = self._get_subscription_id()
+        self.logger.debug(
+            'subscription {0} id is: {1}'.format(sub_name, self.sub_id))
+
+    def _get_subscription_id(self):
+        """Get subscription pool id"""
+        result = Subscription.list(
+            {'organization-id': self.org_id},
+            per_page=False
+        )
+
+        if result.return_code == 0:
+            subscription_id = result.stdout[0]['id']
+            subscription_name = result.stdout[0]['name']
+            self.logger.info('Subscribed to {0} with subscription id {1}'
+                             .format(subscription_name, subscription_id))
+            self.logger.info('---------------------------------')
+            return subscription_id, subscription_name
+        else:
+            self.logger.error('Fail to retrieve subscription id!')
+
+    def test_register_attach_2_clients(self):
+        """
+        @Test: Subscribe system concurrently using 2 virtual machines
+
+        @Steps:
+
+        1. create result dictionary
+        2. concurrent run by multiple threads;
+           each thread iterates a limited number of times
+        3. produce result of timing
+
+        @Assert: Restoring from database without any registered systems.
+        """
+        current_num_threads = self.num_threads[1]
```

Comment:
All test methods could be shorter:

```python
self.kick_off_att_test(self.num_threads[1], 5000)
```

You can drop single use variable without any problem.